lib/tasks/sumofus.rake

```diff
@@ -2,50 +2,146 @@ require 'open-uri'
+    else
+      page_image_handle = open(args[:page_img_file])
     end
 
-    if ARGV[2].nil?
+    if args[:page_img_file].blank?
       abort('Requires a valid url to a file containing a default header image to attach to the page.')
+    else
+      puts "Loading page data"
+      page_data_handle = open(args[:action_file])
+      page_data = JSON.load(page_data_handle.read)
+      page_data_handle.close
+      puts "Page data loaded"
+    end
+
+    if args[:follow_img_file].blank?
+      follow_image_handle = open(args[:page_img_file])
+    else
+      follow_image_handle = open(args[:follow_img_file])
     end
 
-    file_handle = open(ARGV[1])
-    data = file_handle.read
-    file_handle.close
+    def create_post_action_pages(layout_id, image_handle)
+      pages = {}
+      ['en', 'fr', 'de'].map do |locale|
+        page = Page.find_or_initialize_by(title: I18n.t('fundraiser.generic.title', locale: locale))
+        page.liquid_layout_id = layout_id
+        page.language_id = language_ids[locale]
+        page.content = I18n.t('fundraiser.generic.body', locale: locale)
+        page.save!
+        page.images.create!(content: image_handle) if page.images.empty?
+        pages[locale] = page
+      end
+      pages
+    end
 
-    image_handle = open(ARGV[2])
+    def manage_newlines(content)
+      content.
+        gsub(/(?:\n\r?|\r\n?)/, '<br>').
+        gsub(/<br *\/*>/, '<br>').
+        gsub(/<div><br>\t&nbsp;<\/div>/, '<br>').
+        gsub(/(<br>)*\s*<\/p>\s*(<br>)*\s*<p>\s*(<br>)*/, '</p><br><p>').
+        gsub(/(<br>)*\s*<\/div>\s*(<br>)*\s*<div>\s*(<br>)*/, '</p><br><p>').
+        gsub(/(\s*<br>\s*){3,}/, '<br><br>')
+    end
 
-    parsed_data = JSON.load data
-    count = 0
-    layout_id = LiquidLayout.where(title: 'Petition With Small Image').first.id
-    parsed_data.each_pair do |k, entry|
-      count +=1
+    def language_ids
+      return @language_ids unless @language_ids.blank?
+      @language_ids = {}
+      Language.all.each do |l|
+        @language_ids[l.code] = l.id
+      end
+      @language_ids
+    end
+
+    def clean_title(entry)
       title = entry['title'].blank? ? entry['petition_ask'] : entry['title']
-      page = Page.find_or_create_by!(title: title, liquid_layout_id: layout_id)
-      page.content = entry['page_content'].gsub(/(?:\n\r?|\r\n?)/, '<br>')
-      Page.reset_counters(page.id, :actions)
-      Page.update_counters(page.id, action_count: entry['signature_count'])
+      title.chomp(' ').chomp('!').chomp('.')
+    end
+
+    def duplicate_titles(titles)
+      titles.map do |title, variations|
+        variations.size > 1 ? title : nil
+      end.compact
+    end
+
+    def unique_titles(page_data)
+      titles = Hash.new({})
+      # byebug
+      page_data.each_pair do |k, entry|
+        title = clean_title(entry)
+        slug = entry['slug']
+        case titles[title].size
+        when 1
+          titles[title][slug] = "#{title} now"
```

Comment:
basically, if the title is already used, append 'now' to it, then when that's also taken, append 'now!' to it, and when thats taken, append 'today' to it